### ISSUE TITLE
Adjust the color contrast based on the background color

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#589]
**Describe the bug**
The chart is currently rendered on a transparent background. We have two main themes, one for a light background and one for a dark one.
The issue comes out when the user wants to apply a different color to the background: the existing themes are no longer valid in every case for an AA accessibly point of view.
Moreover, the legend is rendered in html that means that it only currently reflect the colors applied through importing the correct CSS method.

A possible solution is:
- apply colors of the chart only through the `Theme` js object, including the text color of the legend 
- configure the background color through a `Settings` prop, this will allow us to tweaks the colors for a better contrast, and also return the current WCAG accessibility level if required.

**To Reproduce**
1. Go to the playground
2. Manually change the background color of the `chart` element in the playground css 
3. Apply the light theme or the dark one
4. The text contrast is not sufficiently high with every background color applied

**Expected behavior**
The light and dark theme should be revisited as a semantic theme, that applies in all possible cases adapting the colors to the background.
A similar approach should be applied to https://github.com/elastic/elastic-charts/issues/238 

**Screenshots**
with light theme:
<img width="271" alt="Screenshot 2020-03-17 at 10 52 15" src="https://user-images.githubusercontent.com/1421091/76844383-7c8b0200-683d-11ea-970f-0852c986e73f.png">

with dark theme:
<img width="260" alt="Screenshot 2020-03-17 at 10 52 38" src="https://user-images.githubusercontent.com/1421091/76844387-7dbc2f00-683d-11ea-9883-519907fae625.png">

**Kibana Cross Issues**
- [ ] https://github.com/elastic/kibana/issues/60011

**Checklist**
- [x] every related Kibana issue is listed under `Kibana Cross Issues` list
- [x] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

**Comment by rshen91** [comment]
I'm wondering if this issue might be resolved with #608?

**Comment by markov00** [comment]
I don't think so, @rshen91 can you test this to see if it happens on master?

**Comment by nickofthyme** [comment]
Just to clarify, the idea here is to apply color contrast logic to all the colors, not only text right. So that would include axes lines, tick and labels as well as the series colors themselves?

**Comment by markov00** [comment]
Sorry I've missed that reply:
@nickofthyme right, we have to apply the color contrast logic when the user applies a specific background color to all static elements in charts axis, ticks, labels, legend text etc. This because we have 2 themes defined for dark and light background but don't cover automatically the issues when using a dark background color on the chart when using a light background color in Kibana. In this particular example, the legend item text reflects the CSS style used in the global Kibana theme

**Comment by markov00** [comment]
I believe we should close this as not planned. There will be always something that we can't control or that will override this contrast improvement.
What we can provide in the future could be something like a utility to generate or check theme contrast rules, but I believe we should not automatically change theme colors depending on chart backgrounds.
